graylog2-server/src/main/java/org/graylog2/grok/MongoDbGrokPatternService.java

```diff
@@ -50,11 +47,10 @@
 import java.util.regex.PatternSyntaxException;
 import java.util.stream.Collectors;
 
-import static com.google.common.base.Preconditions.checkNotNull;
 import static org.graylog2.grok.GrokPatternService.ImportStrategy.ABORT_ON_CONFLICT;
 import static org.graylog2.grok.GrokPatternService.ImportStrategy.DROP_ALL_EXISTING;
 
```

Comment:
I think this could be made a singleton to avoid creating the index every time the class is constructed (i.e. during pagination)

```suggestion
@Singleton
```

Reply:
Mongo doesn't recreate an existing index. But it's still a good idea to avoid an unnecessary roundtrip.